### PR TITLE
Readd Nested filter

### PR DESCRIPTION
**Pull request by ruflin**
See https://github.com/ruflin/Elastica/issues/1001

### CODE REVIEW

**Comment by ruflin** [comment]
@kukulich Because of the tests that are broken I actually prefer to keep this PR currently out of 3.0.0. It is easy to add a feature later but removing and changing BC is worse I think. As we plan to ship 3.1 as soon as this is fixed, I don't think it should be a big issue.

**Comment by kukulich** [comment]
@ruflin I don't know if I understand you correctly. Do you plan to release 3.0.0 without `Filter\Nested`?

**Comment by ruflin** [comment]
@kukulich Yes, I just did. The reason is that it is very easy to add code after a release but hard to remove it. The tests for nested didn't pass because some results were empty instead of results. So I prefer that people are aware of the issue instead of just using it and not getting results. I hope to ship either 3.0.1 or 3.1.0 in the next days with a fix.

File: test/lib/Elastica/Test/Filter/NestedFilterWithSetFilterTest.php [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+    /**
+     * @group functional
+     */
+    public function testShouldReturnTheRightNumberOfResult()
+    {
+        $filter = new Nested();
+        $this->assertEquals(array('nested' => array()), $filter->toArray());
+        $query = new Terms();
+        $query->setTerms('hobby', array('guitar'));
```
Comment (kukulich) [inline test/lib/Elastica/Test/Filter/NestedFilterWithSetFilterTest.php]: @ruflin From the documentation "Note that any fields referenced inside the query must use the complete path (fully qualified)."

This should work `$query->setTerms('hobbies.hobby', array('guitar'));`

Comment (ruflin) [inline test/lib/Elastica/Test/Filter/NestedFilterWithSetFilterTest.php]: @kukulich that is great. Seems to work.

**Comment by ruflin** [comment]
@ewgRa Seems like @kukulich just provided a very simple way to fix this PR. I suggest to merge it and release 3.0.1 with it. What do you think?

**Comment by ewgRa** [comment]
Do it, I will merge it to my branch, I think it is not a problem for me.

**Comment by ruflin** [comment]
@ewgRa Perfect. Lets wait for it to get green.

**Comment by ruflin** [comment]
@kukulich Merged. Thanks a lot for the fast feedback for the fix.

**Comment by kukulich** [comment]
You're welcome. I'm happy that `Filter\Nested` returned :)

**Pull request merged**